File: src/codes/scattered-concerns/py/rfc_facade.py
import os

class Config:
    @staticmethod
    def is_debug() -> bool:
        return os.environ.get("DEBUG_MODE") == "true"

class LoggerFacade:
    @staticmethod
    def log(message: str):
        print(f"[LOG] {message}")

    @staticmethod
    def debug(message: str):
        if Config.is_debug():
            print(f"[DEBUG] {message}")

class OrderService:
    def create_order(self, order_id: str):
        LoggerFacade.log(f"注文作成: {order_id}")
        LoggerFacade.debug(f"注文詳細: {order_id}")

File: src/codes/scattered-concerns/php/rfc_facade.php
<?php

class Config {
  public static function isDebug(): bool {
    return getenv("DEBUG_MODE") === "true";
  }
}

class LoggerFacade {
  public static function log(string $message): void {
    echo "[LOG] {$message}\n";
  }

  public static function debug(string $message): void {
    if (Config::isDebug()) {
      echo "[DEBUG] {$message}\n";
    }
  }
}

class OrderService {
  public function createOrder(string $orderId): void {
    LoggerFacade::log("注文作成: {$orderId}");
    LoggerFacade::debug("注文詳細: {$orderId}");
  }
}
